judo-runtime-core-dispatcher/src/main/java/hu/blackbelt/judo/runtime/core/dispatcher/DefaultDispatcher.java

```diff
@@ -520,227 +522,233 @@ private void setupValidationContextForBoundOperation(AsmUtils.OperationBehaviour
-            entityType = Optional.of(getAsmUtils().resolve(signedIdentifier.getEntityType()).filter(t -> t instanceof EClass).map(t -> (EClass) t)
+            entityType = Optional.of(asmUtils.resolve(signedIdentifier.getEntityType()).filter(t -> t instanceof EClass).map(t -> (EClass) t)
                     .orElseThrow(() -> new IllegalArgumentException("Unable to resolve entity type")));
             final Payload metadata = dao.getMetadata(mappedTransferObjectType, dataTypeManager.getCoercer().coerce(signedIdentifier.getIdentifier(), identifierProvider.getType()))
                     .orElseThrow(() -> new NotFoundException(ValidationResult.builder()
                             .code(ERROR_BOUND_OPERATION_INSTANCE_NOT_FOUND)
                             .level(ValidationResult.Level.ERROR)
                             .build()));
-            checkState(Objects.equals(AsmUtils.getClassifierFQName(entityType.get()), metadata.get(Dispatcher.ENTITY_TYPE_MAP_KEY)), "Invalid entity type in signed identifier");
+            checkState(Objects.equals(getClassifierFQName(entityType.get()), metadata.get(Dispatcher.ENTITY_TYPE_MAP_KEY)), "Invalid entity type in signed identifier");
         } else {
             final Payload metadata = getTransferObjectAsBoundType(mappedTransferObjectType, signedIdentifier);
             final String entityTypeFQName = metadata.getAs(String.class, ENTITY_TYPE_MAP_KEY);
             checkArgument(entityTypeFQName != null, "Entity type is unknown");
 
-            entityType = Optional.of(getAsmUtils().resolve(entityTypeFQName).filter(t -> t instanceof EClass).map(t -> (EClass) t)
+            entityType = Optional.of(asmUtils.resolve(entityTypeFQName).filter(t -> t instanceof EClass).map(t -> (EClass) t)
                     .orElseThrow(() -> new IllegalArgumentException("Unable to resolve entity type")));
         }
         return entityType;
     }
 
+    CacheLoader<String, Optional<EOperation>> operationLoader = new CacheLoader<>() {
+        @Override
+        public Optional<EOperation> load(String operationFullyQualifiedName) {
+            return asmUtils.all(EOperation.class)
+                    .filter(op -> Objects.equals(getOperationFQName(op), operationFullyQualifiedName))
+                    .findAny();
+        }
+    };
+
+    private LoadingCache<String, Optional<EOperation>> operationCache = CacheBuilder.newBuilder().build(operationLoader);
+
     @Override
     public Map<String, Object> callOperation(final String operationFullyQualifiedName, final Map<String, Object> exchange) {
         MDC.put("operation", operationFullyQualifiedName);
         Payload result = null;
 
         final long startTs = System.nanoTime();
+        final boolean exposed = Boolean.TRUE.equals(exchange.remove("__exposed"));
 
+        final Boolean stateful = context.getAs(Boolean.class, STATEFUL);
+        // TODO - do not cleanup context of operations called by dispatcher
```

Comment:
TODO in what sense, de we need a follow-up ticket?

Reply:
This message was not added by me. :)